lib/keen-query.js

```diff
@@ -227,32 +227,56 @@ class KeenQuery {
 		return this;
 	}
 
-	generateKeenUrl (base) {
-		base = typeof base === 'undefined' ? `https://api.keen.io/3.0/projects/${configContainer.KEEN_PROJECT_ID}/queries/${this.extraction}?api_key=${configContainer.KEEN_READ_KEY}`: base;
+	// format can be `api` or `explorer`
+	generateKeenUrl (format) {
+		format = typeof format === 'undefined' ? 'api' : format
+
+		let properties = {
+			analysis_type: this.extraction,
+			target_property: this.query.target_property,
+			event_collection: this.query.event_collection,
+			interval: this.query.interval,
+			timeframe: this.timeframe,
+			group_by: this.groupedBy
+		}
 
-		const parts = [base];
-		const filters = this.filters.slice();
+		// Todo: Move this excludeNulls logic outside of generateKeenUrl()
+		if (this.groupedBy.length && this.excludeNulls) {
+			if (!this.filters) this.filters = []
+
+			this.groupedBy.forEach(prop => {
+				filters.push({
+					property_name: prop,
+					operator: 'exists',
+					property_value: true
+				});
+			})
+		}
 
-		if (this.groupedBy.length) {
-			parts.push(`group_by=${this.groupedBy.length === 1 ? this.groupedBy[0] : encodeURIComponent(JSON.stringify(this.groupedBy))}`);
-
-			if (this.excludeNulls) {
-				this.groupedBy.forEach(prop => {
-					filters.push({
-						property_name: prop,
-						operator: 'exists',
-						property_value: true
-					});
-				})
+		if (format === 'explorer') {
+			let explorerParts = [].concat(Object.keys(properties)
+				.filter(p => properties[p] && (typeof properties[p] === 'object' ? properties[p].length : properties[p]))
+				.map(p => `query[${p}]=${encodeURIComponent(properties[p])}`))
+
+			const filterParts = this.filters.map((f, i) => Object.keys(f).map(p => `query[filters][${i}]=${encodeURIComponent(f[p])}`))[0];
+			if (filterParts && filterParts.length) {
+				explorerParts = explorerParts.concat(filterParts)
 			}
+
+			return `/data/explorer?${explorerParts.join('&')}`;
```

Comment:
This is a next specific url so I'd like to keep this out of keen-query.

I think the way to tackle this is to have 2 params `(format, baseUrl)`

then
```
if (format === 'api') {
   baseUrl = baseUrl || 'http://api.keen.io...'
} else {
 baseUrl = baseUrl || ''
}
```